src/ptvsd/adapter/messages.py

```diff
@@ -135,8 +137,51 @@ def initialize_request(self, request):
     # Handles various attributes common to both "launch" and "attach".
     def _debug_config(self, request):
         assert request.command in ("launch", "attach")
-        pass  # TODO: options and debugOptions
-        pass  # TODO: pathMappings (unless server does that entirely?)
+        args = request.arguments
+        self.debug_options = dbg_options.extract_debug_options(
+            args.get("options"), args.get("debugOptions")
+        )
+
+        client_os_type = self.debug_options.get("CLIENT_OS_TYPE", "").upper().strip()
+        if bool(client_os_type) and client_os_type not in ("WINDOWS", "UNIX"):
+            ptvsd.log.warn(
+                'Invalid CLIENT_OS_TYPE passed: %s (must be either "WINDOWS" or "UNIX").'
+                % (client_os_type,)
+            )
```

Comment:
Just `log.warn`, since it's imported via `from`.

Also, `warn` does string formatting by itself - its first argument is a format string, of the `"{0}"` variety.